hack/generate-kustomize-patches.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# This script will generate kustomize patches for webhooks by first getting the
+# MutatingWebhookConfiguration and ValidatingWebhookConfiguration, so it
+# generates patches only or those webhooks that are specified in upstream.
+#
+# This way we don't have to specify webhook patches manually, since those
+# change from time to time, as CRDs are added, modified and removed.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Directories
+ROOT_DIR="./$(dirname "$0")/.."
+ROOT_DIR="$(realpath "$ROOT_DIR")"
+KUSTOMIZE_DIR="$ROOT_DIR/config/helm"
+HELM_DIR="$ROOT_DIR/helm/cluster-api-provider-azure"
+KUSTOMIZE_INPUT_DIR="$ROOT_DIR/config/helm/input"
+
+# Download upstream manifests
+helm_values="$HELM_DIR/values.yaml"
+org="kubernetes-sigs"
+repo="cluster-api-provider-azure"
+version="$(yq e '.image.tag' "$helm_values")"
+release_asset_filename="infrastructure-components.yaml"
+url="https://github.com/$org/$repo/releases/download/$version/$release_asset_filename"
+mkdir -p "$KUSTOMIZE_INPUT_DIR"
+curl -L "$url" -o "$KUSTOMIZE_INPUT_DIR/$release_asset_filename"
+
+# Update kustomize patches for webhooks. We do this for every CRD
+
+# First clear previous watchfilter patches
+true > "$KUSTOMIZE_DIR/webhook-watchfilter.yaml"
+watch_filter="$(yq e '.watchfilter' "$helm_values")"
```

Comment:
what about hardcoding the `watch_filter` label? We won't change it, so why making it configurable?

Reply:
We had it in Helm `values.yaml` in CAPI app I think, so I just aligned other apps with that. But yeah, we can just hard-code it everywhere.